runtime/android/core/src/org/xwalk/core/XWalkActivity.java

```diff
@@ -128,4 +188,15 @@ protected void onResume() {
         super.onResume();
         mActivityDelegate.onResume();
     }
+
+    private boolean isDownloadMode() {
```

Comment:
Please keep `XWalkActivity` as simple as possible and move all specific process to `XWalkActivityDelegate`, because Cordova plugin wants to share same functions with `XWalkActivity` through `XWalkActivityDelegate`.